meerk40t/gui/icons.py

```diff
@@ -569,216 +586,207 @@ def GetBitmap(
+                else:
+                    sbrush = wx.Brush()
+                    sbrush.SetColour(colpat)
+                gc.SetBrush(sbrush)
+                gc.FillPath(gp, fillStyle=fill_style)
+            for entry in stroke_paths:
+                gp = entry[0]
+                attrib = entry[2]
+                if entry[1] is None:
+                    spen = self._pen
+                else:
+                    spen = wx.Pen()
+                    spen.SetColour(entry[1])
+                    spen.SetWidth(self.strokewidth)
+                spen.SetCap(wx.CAP_ROUND)
+                if "cap_butt" in attrib:
+                    spen.SetCap(wx.CAP_BUTT)
+                if "cap_round" in attrib:
+                    spen.SetCap(wx.CAP_BUTT)
+                if "cap_square" in attrib:
+                    spen.SetCap(wx.CAP_PROJECTING)
 
-            if colpat is None:
-                sbrush = self._brush
-            else:
-                sbrush = wx.Brush()
-                sbrush.SetColour(colpat)
-            gc.SetBrush(sbrush)
-            gc.FillPath(gp, fillStyle=fill_style)
-        for entry in stroke_paths:
-            gp = entry[0]
-            attrib = entry[2]
-            if entry[1] is None:
-                spen = self._pen
-            else:
-                spen = wx.Pen()
-                spen.SetColour(entry[1])
-                spen.SetWidth(self.strokewidth)
-            spen.SetCap(wx.CAP_ROUND)
-            if "cap_butt" in attrib:
-                spen.SetCap(wx.CAP_BUTT)
-            if "cap_round" in attrib:
-                spen.SetCap(wx.CAP_BUTT)
-            if "cap_square" in attrib:
-                spen.SetCap(wx.CAP_PROJECTING)
-
-            spen.SetJoin(wx.JOIN_ROUND)
-            if "join_arcs" in attrib:
                 spen.SetJoin(wx.JOIN_ROUND)
-            if "join_bevel" in attrib:
-                spen.SetJoin(wx.JOIN_BEVEL)
-            if "join_miter" in attrib:
-                spen.SetJoin(wx.JOIN_MITER)
-            if "join_miterclip" in attrib:
-                spen.SetJoin(wx.JOIN_MITER)
-            gc.SetPen(spen)
-            gc.StrokePath(gp)
-        dc.SelectObject(wx.NullBitmap)
-        gc.Destroy()
-        del gc.dc
-        del dc
-        # Save bitmap for later retrieval
-        _CACHE[cache_id] = bmp
+                if "join_arcs" in attrib:
+                    spen.SetJoin(wx.JOIN_ROUND)
+                if "join_bevel" in attrib:
+                    spen.SetJoin(wx.JOIN_BEVEL)
+                if "join_miter" in attrib:
+                    spen.SetJoin(wx.JOIN_MITER)
+                if "join_miterclip" in attrib:
+                    spen.SetJoin(wx.JOIN_MITER)
+                gc.SetPen(spen)
+                gc.StrokePath(gp)
+            dc.SelectObject(wx.NullBitmap)
+            gc.Destroy()
+            del gc.dc
+            del dc
+            # Save bitmap for later retrieval
+            _CACHE[cache_id] = bmp
```

Comment:
Due to the extra tab, I can't tell if anything changed. I assume not.

Reply:
Removed the local caching scheme in favor of the functools lru-cache functionality.